### DISCUSSION TITLE
Use DateTime<Utc> for Postgres TIMESTAMPTZ

### DISCUSSION THREAD

**Discussion by tqwewe** [SeaQL/sea-orm#68] [Q&A] [answered]
I have a model with a DB column called `created_at` which is of type `TIMESTAMPTZ`.
My model uses the type `DateTime<Utc>` for the timestamp, but I get an error saying:
```
   | #[derive(Clone, Debug, PartialEq, DeriveModel, DeriveActiveModel)]
   |                                                ^^^^^^^^^^^^^^^^^ the trait `From<chrono::DateTime<Utc>>` is not implemented for `sea_orm::Value`
 pub struct Model {
     pub id: Uuid,
     pub created_at: DateTime<Utc>,
 }
```
I have the `with-chrono` flag enabled.

How should I be querying a `TIMESTAMPTZ` column? I tried making the Rust type just a `String` but get an error when querying saying:
> Error: Query Error: error occurred while decoding column "created_at": mismatched types; Rust type `alloc::string::String` (as SQL type `TEXT`) is not compatible with SQL type `TIMESTAMPTZ`

**Comment by tyt2y3** [comment]
`TIMESTAMPTZ` is not yet supported. But AFAIK `DateTime<Utc>` is equivalent to `NaiveDateTime` which is aliased to `DateTime` in SeaORM

**Reply by tqwewe** [reply]
Okay I'd love for SeaORM to support `TIMESTAMPTZ`, I'll look into the code to see if I can figure it out

**Reply by tyt2y3** [reply]
SeaQuery had basic support for `DateTimeWithTimeZone`, but that did not extend to SeaORM yet

**Comment by tyt2y3** [comment, marked as answer]
I added support for the type `chrono::DateTime<chrono::FixedOffset>` which is aliased to `DateTimeWithTimeZone` in SeaORM.
#71

**Reply by tqwewe** [reply]
Legend! I'll pull down master branch and try it out

**Reply by tqwewe** [reply]
@tyt2y3 I tried with the latest code from master with:
```rust
#[derive(Clone, Debug, PartialEq, DeriveModel, DeriveActiveModel)]
pub struct Model {
    pub id: Uuid,
    pub created_at: DateTimeWithTimeZone,
```
And with:
```rust
#[derive(Clone, Debug, PartialEq, DeriveModel, DeriveActiveModel)]
pub struct Model {
    pub id: Uuid,
    pub created_at: DateTime<Utc>,
```
But both give me an error for `DeriveActiveModel`:
```
the trait `From<chrono::DateTime<FixedOffset>>` is not implemented for `sea_orm::Value`
```

**Reply by tyt2y3** [reply]
I think you should `cargo update` to update sea-query to `0.12.9` too.

**Reply by tqwewe** [reply]
That seems to have worked!

**Reply by brecht-derooms** [reply]
shouldn't this be chrono::DateTime<Utc> rather?
At least sqlx expects chrono::DateTime<Utc> so it makes the model incompatible if you use it both with sqlx as with seaorm. 
Timestamp with timezone in postgres is notoriously confusing, it's however the recommended type to store utc timestmaps as well as far as I recall.

**Comment by marlon-sousa** [comment]
Hello,

I have kind of the same problem but need orientation to solve that.

## First things first

1. My postgres table has a column called created_at with type "timestamp with time zone"
2. I use sea-orm-cli to generate entity (set DATABASE_URL environment variable then sea-orm-cli generate entity -o some_dir)
3. I notice that the columns generated have the NaiveDateTime type (I see pub created_at: DateTime, declared on the generated model).

## Environment

* sea-cli-orm v 2.0.0
* In my project, I have sea-orm = { version = "^0", features = ["sqlx-postgres", "runtime-tokio-native-tls", "macros"], default-features = false }

## Error


Queries run show the following error:

mismatched types; Rust type `core::option::Option<chrono::naive::datetime::NaiveDateTime>` (as SQL type `TIMESTAMP`) is not compatible with SQL type `TIMESTAMPTZ`"

If I try to use DateTime<UTC> (by changing the generated entities by hand) I have a trait error.

Without waiting for the cli to correctly recognize the type (if it isn't still doing so), how can I fix that?

Thanks,
Marlon

**Comment by tyt2y3** [comment]
It should be DateTimeWithTimeZone

https://github.com/SeaQL/sea-orm/blob/master/src/entity/prelude.rs
________________________________

**Comment by marlon-sousa** [comment]
Hello,

Changing to DateTimeWithTimeZone worked, at least somehow.

I see that this translates to ``` chrono::DateTime<FixedOffset> ```. How does this relate to ``` chrono::DateTime<Utc> ```? I ask this because I kit francly couldn't find the difference and the way these columns are treatened by other orms is using Utc...

**Comment by JohnKinyanjui** [comment]
Hello not sure if it will wot=rk for anyone but worked for me
```
struct Test{
    pub created_at: Option<chrono::NaiveDateTime>,
}
```
